internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/PushStatusDecider.java

```diff
@@ -251,42 +252,29 @@ private void logConditionally(String newReason, Optional<String> previousReason,
   protected ExecutionStatus getPartitionStatus(
       PartitionStatus partitionStatus,
       int replicationFactor,
-      Map<Instance, String> instanceToStateMap,
-      DisableReplicaCallback callback) {
-    return getPartitionStatus(
-        partitionStatus,
-        replicationFactor,
-        instanceToStateMap,
-        getNumberOfToleratedErrors(),
-        callback);
-  }
-
-  protected ExecutionStatus getPartitionStatus(
-      PartitionStatus partitionStatus,
-      int replicationFactor,
-      Map<Instance, String> instanceToStateMap,
-      int numberOfToleratedErrors,
+      Map<Instance, HelixState> instanceToStateMap,
       DisableReplicaCallback callback) {
-    Map<ExecutionStatus, Integer> executionStatusMap = new HashMap<>();
+    int numberOfToleratedErrors = getNumberOfToleratedErrors();
+    Map<ExecutionStatus, Integer> executionStatusMap = new EnumMap<>(ExecutionStatus.class);
 
     // when resources are running under L/F model, leader is usually taking more critical work and
     // are more important than followers. Therefore, we strictly require leaders to be completed before
     // partitions can be completed. Vice versa, partitions will be in error state if leader is in error
     // state.
     boolean isLeaderCompleted = true;
     int previouslyDisabledErrorReplica = 0;
-    for (Map.Entry<Instance, String> entry: instanceToStateMap.entrySet()) {
+    for (Map.Entry<Instance, HelixState> entry: instanceToStateMap.entrySet()) {
       ExecutionStatus currentStatus =
           getReplicaCurrentStatus(partitionStatus.getReplicaHistoricStatusList(entry.getKey().getNodeId()));
-      if (entry.getValue().equals(HelixState.LEADER_STATE)) {
+      if (entry.getValue() == HelixState.LEADER) {
```

Comment:
Hmm, I thought .equals() should be the better way?

Reply:
enum `==` projects against NPE.

Reply:
Enum values are singletons, so we can do `==` on them, just like with primitive.

I prefer doing `==` whenever possible since it is more concise (and for me, easier to read), it gets compile-time type safety (whereas you could call `Enum1.BLAH.equals(Enum2.BLEH)` and technically it compiles), and you can't get NPEs on `==`.